lib/super_digit.rb

```diff
@@ -1,15 +1,28 @@
 # Superdigit
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(n+logn) n being number of digits, because sum method takes a O(n) time, and the recusrive method call on a worst case scenario like superdigit(1296) will cut the input in half when you call the recursive method superdigit(18) so logn
+# Space Complexity - O(n) n being number of digits in a number
 def super_digit(n)
```

Comment:
Since you are doing the num_array.sum each time (and doing n.to_s.map as well) you're doing an O(n) operation log n times by your logic, so it wouldn't be O(n + log n)